### PR TITLE
fix ripper projectile when blockmap fix is enabled

### LINKED ISSUE #1971: Improved Hit Detection makes ripper projectiles deal damage multiple times
_Tested in the [latest master artifact (Win-64)](https://github.com/fabiangreffrath/woof/actions/runs/11621341661)._

As the title says: with the _Improved Hit Detection_ setting enabled, ripper projectiles deal damage multiple times per tic -- a bug made noticeable to me by the fact that multiple blood splats are spawned together.

I tested this with ID24's Calamity Blade; it probably has to do with both the projectile and the target overlapping with multiple blocks.

### PR DESCRIPTION
**Pull request by rfomin**
Fix #1971

### CODE REVIEW

File: src/p_maputl.c [hunk 1]
```diff
@@ -444,6 +444,9 @@ boolean P_BlockThingsIterator(int x, int y, boolean func(mobj_t*))
     if (func == PIT_RadiusAttack || func == PIT_ChangeSector)
       return true;
 
+    if (mobj->flags2 & MF2_RIP)
```
Comment (fabiangreffrath) [inline src/p_maputl.c:447]: Don't we have to check the `tmthing`'s flags?

Comment (rfomin) [reply src/p_maputl.c]: Right, this is wrong approach.

**Comment by fabiangreffrath** [comment]
We could pass `tmthing` as another parameter to `P_BlockThingsIterator()`.

**Comment by rfomin** [comment]
> We could pass `tmthing` as another parameter to `P_BlockThingsIterator()`.

Similar approach: https://github.com/fabiangreffrath/woof/pull/1977

**Pull request closed without merging**